tool/tsh/tsh.go

```diff
@@ -454,6 +454,23 @@ func setupNoninteractiveClient(tc *client.TeleportClient, key *client.Key) error
 		return trace.Wrap(err)
 	}
 	tc.Username = certUsername
+
+	// Extract and set the HostLogin to be the first principal. It doesn't
+	// matter what the value is, but some valid principal has to be set
+	// otherwise the certificate won't be validated.
+	publicKey, _, _, _, err := ssh.ParseAuthorizedKey(key.Cert)
```

Comment:
why not factor this into something similar to `CertUsername` in Key's method list? E.g. `CertLogin` will look good